src/OAuth2Form.tsx

```diff
@@ -1,403 +1,439 @@
+import { getStateParam, getURLParam, handleErrorRedirect } from './URLUtils'
+import { Paper, Button, Link } from '@mui/material'
+import { StyledInnerContainer } from 'StyledInnerContainer'
 
 export const OAuth2Form = () => {
-  const isMounted = useRef(true);
+  const isMounted = useRef(true)
 
-  const { accessToken, setAccessToken } = useOAuthAppContext();
-
-  const [error, setError] = useState<any>();
+  const { accessToken, setAccessToken } = useOAuthAppContext()
+  const [isSessionEstablished, setIsSessionEstablished] =
+    React.useState<boolean>(false)
+  const [error, setError] = useState<any>()
 
   const onError = useCallback(
     (error: Error | OAuthClientError | SynapseClientError) => {
-      debugger;
+      debugger
       if (error instanceof SynapseClientError && error.status === 401) {
         // invalid token, so clear it
-        SynapseClient.signOut(() => {
-          setAccessToken(undefined);
-        });
+        SynapseClient.signOut().then(() => {
+          setAccessToken(undefined)
+        })
       } else {
-        handleErrorRedirect(error);
-        setError(error);
+        handleErrorRedirect(error)
+        setError(error)
       }
     },
-    [setAccessToken]
-  );
+    [setAccessToken],
+  )
 
   // In addition to fetching the current user profile, the success of this request will determine if the current access token is valid.
   const { data: profile } = useGetCurrentUserProfile({
     enabled: !!accessToken,
     onError,
-  });
+  })
 
   if (profile?.profilePicureFileHandleId) {
-    profile.clientPreSignedURL = `https://www.synapse.org/Portal/filehandleassociation?associatedObjectId=${profile.ownerId}&associatedObjectType=UserProfileAttachment&fileHandleId=${profile.profilePicureFileHandleId}`;
+    profile.clientPreSignedURL = `https://www.synapse.org/Portal/filehandleassociation?associatedObjectId=${profile.ownerId}&associatedObjectType=UserProfileAttachment&fileHandleId=${profile.profilePicureFileHandleId}`
   }
 
   const [oidcRequestDescription, setOidcRequestDescription] =
-    useState<OIDCAuthorizationRequestDescription>();
-  const [oauthClientInfo, setOauthClientInfo] = useState<OAuthClientPublic>();
-  const [redirectURL, setRedirectURL] = useState<string>();
-
-  const [isConsenting, setIsConsenting] = useState<boolean>(false);
+    useState<OIDCAuthorizationRequestDescription>()
+  const [oauthClientInfo, setOauthClientInfo] = useState<OAuthClientPublic>()
+  const [redirectURL, setRedirectURL] = useState<string>()
+  const [isPreviousAuthCheckComplete, setIsPreviousAuthCheckComplete] =
+    useState<boolean>(false)
+  const [isConsenting, setIsConsenting] = useState<boolean>(false)
 
   useEffect(() => {
-    isMounted.current = true;
+    isMounted.current = true
     return () => {
-      isMounted.current = false;
-    };
-  });
+      isMounted.current = false
+    }
+  })
+
+  useEffect(() => {
+    const getSession = async () => {
```

Comment:
This function was moved from elsewhere.  Now the Standalone Login Form will end up setting isSessionEstablished to true which causes this to execute (updating the client access token)

Reply:
Not seeing `isSessionEstablished` used anywhere else. Could we just do:
```tsx
const getSession = useCallback(async () => {
  // ...
}, [setAccessToken])
```
And have the `sessionCallback` prop just invoke `getSession`?

Reply:
That's much better - I made this change